networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/core/encodings/AbstractByteBufDecoder.java

```diff
@@ -30,12 +30,16 @@
     implements ByteBufDecoder<TMessage, TException> {
 
   private CompositeByteBuf compositeByteBuf = Unpooled.compositeBuffer();
+  private boolean closed = false;
 
   @Override
   public Optional<TMessage> decodeOneMessage(ByteBuf in) throws TException {
     if (!in.isReadable()) {
       return Optional.empty();
     }
+    if (closed) {
+      throw new IllegalStateException("Trying to reuse disposed decoder instance");
+    }
```

Comment:
(nit / optional) We could move this to a helper method `assertOpen()`

Reply:
Yep good idea  👍  Just named it assertNotClosed since Opened sounds a bit weird with respect to Decoder :)